### DISCUSSION TITLE
how to use "--record-and-pipe"?

### DISCUSSION THREAD

**Discussion by raekuul** [streamlink/streamlink#5951] [Q&A] [answered]
### Description

I'm trying to use the `--record-and-pipe` flag to both save a stream to disk and watch it as it's being saved (the cli invocation is `streamlink <url> <quality> --record-and-pipe "./test.ts" | ffplay -i -`, but it's throwing a "cannot use record options with other file output options" error.  Do I just have the options in the wrong order, do I need more/less/different options, or is this actually an option that cannot be used?

**Comment by bastimeyer** [comment, marked as answer]
It works exactly as described here:
https://streamlink.github.io/cli.html#cmdoption-record-and-pipe

Are you by accident loading a config file which sets other arguments?
https://streamlink.github.io/cli/config.html

Post the `--loglevel=debug` output.

```
$ streamlink --record-and-pipe /dev/null twitch.tv/eslcs best | mpv -
[file] Reading from stdin...
[cli][info] Found matching plugin twitch for URL twitch.tv/eslcs
[cli][info] Available streams: audio_only, 160p (worst), 360p, 480p, 720p, 720p60, 1080p60 (best)
[cli][info] Opening stream: 1080p60 (hls)
[cli][info] Writing output to
/dev/null
[plugins.twitch][info] Will skip ad segments
[plugins.twitch][info] Low latency streaming (HLS live edge: 2)
[plugins.twitch][info] This is not a low latency stream
 (+) Video --vid=1 (h264 1920x1080 60.000fps)
 (+) Audio --aid=1 (aac 2ch 48000Hz)
[download] Written 1.69 MiB to /dev/null (0s)                                   AO: [pulse] 48000Hz stereo 2ch float
VO: [gpu] 1920x1080 yuv420p
[download] Written 6.10 MiB to /dev/null (5s @ 1.28 MiB/s)                      Exiting... (Quit)
[download] Written 6.10 MiB to /dev/null (5s @ 1.08 MiB/s)                      
[cli][info] Stream ended
error: Error when writing to output: [Errno 32] Broken pipe, exiting
[cli][info] Closing currently open stream...
```

**Reply by raekuul** [reply]
Thank you, that got me to find the problem - since this config is reused across multiple machines with different players, I had added "stdout" as one of the options in the main config (and forgotten that I had done so).
